### PR TITLE
specified language cxx in cmake where appropiate

### PR DESCRIPTION
**Pull request by sameergupta4873**
### Description

Specified LANGUAGE CXX in cmake where appropriate per issue #1923 , also I had added 
`project(<pakage name>
  VERSION 1.0.2
  LANGUAGES CXX
)` here specifying VERSION  was important, else the error of lesser arguments provided occurs, I am not sure which version is needed rather than 1.0.2, build is successful locally tough.

### Checklist
- [ ] **Required by CI**: Code is auto formatted using [clang-format](http://moveit.ros.org/documentation/contributing/code)
- [ ] Extend the tutorials / documentation [reference](http://moveit.ros.org/documentation/contributing/)
- [ ] Document API changes relevant to the user in the [MIGRATION.md](https://github.com/ros-planning/moveit/blob/master/MIGRATION.md) notes
- [ ] Create tests, which fail without this PR [reference](https://ros-planning.github.io/moveit_tutorials/doc/tests/tests_tutorial.html)
- [ ] Include a screenshot if changing a GUI
- [ ] While waiting for someone to review your request, please help review [another open pull request](https://github.com/ros-planning/moveit/pulls) to support the maintainers

[//]: # "You can expect a response from a maintainer within 7 days. If you haven't heard anything by then, feel free to ping the thread. Thank you!"

### CODE REVIEW

**Comment by tylerjw** [comment]
I don't think we want to add VERSION this way because the version of the package when distributed as a binary is set by the ros tooling based on package.xml. If we did add the versions here we would need to be careful to update these at the same time as the package.xml. It would be better to have the version of each package just be specified in one place imo.

**Comment by tylerjw** [comment]
Can you explain what you mean by this?

> else the error of lesser arguments provided occurs

@ChrisThrasher do you know what issue the author here is running into?

**Comment by ChrisThrasher** [comment]
There's no need to add the `VERSION` nor am I sure what errors could have occurred by simply adding `LANGUAGES CXX` to the `project()` call.

**Comment by sameergupta4873** [comment]
> There's no need to add the `VERSION` nor am I sure what errors could have occurred by simply adding `LANGUAGES CXX` to the `project()` call.

For instance in [moveit_core](https://github.com/sameergupta4873/moveit2/tree/1a8fa0312010a730aecc9038214a3e366fdc0f23/moveit_core)/[collision_detection](https://github.com/sameergupta4873/moveit2/tree/1a8fa0312010a730aecc9038214a3e366fdc0f23/moveit_core/collision_detection)/CMakeLists.txt in line [38](https://github.com/sameergupta4873/moveit2/blob/1a8fa0312010a730aecc9038214a3e366fdc0f23/moveit_core/collision_detection/CMakeLists.txt#L38), It had VERSION Definition required so we need specify it in here.
Error logs otherwise :-
```--- stderr: moveit_core
CMake Error at collision_detection/CMakeLists.txt:37 (set_target_properties):
  set_target_properties called with incorrect number of arguments.
```

**Comment by ChrisThrasher** [comment]
> For instance in [moveit_core](https://github.com/sameergupta4873/moveit2/tree/1a8fa0312010a730aecc9038214a3e366fdc0f23/moveit_core)/[collision_detection](https://github.com/sameergupta4873/moveit2/tree/1a8fa0312010a730aecc9038214a3e366fdc0f23/moveit_core/collision_detection)/CMakeLists.txt in line [38](https://github.com/sameergupta4873/moveit2/blob/1a8fa0312010a730aecc9038214a3e366fdc0f23/moveit_core/collision_detection/CMakeLists.txt#L38), It had VERSION Definition required so we need specify it in here.

I don't understand. The moveit_core package already specifies `LANGUAGES CXX` and collision_detect still builds just fine. Were you trying to modify that `set_target_properties` call?

File: moveit_core/collision_detection/CMakeLists.txt [hunk 1]
```diff
@@ -1,3 +1,7 @@
+project(moveit_collision_detection
+  VERSION 1.0.2
+  LANGUAGES CXX
+)
```
Comment (ChrisThrasher) [inline moveit_core/collision_detection/CMakeLists.txt:4]: I realize the problem. This is not what we want to do. We don't want to add more `project` calls. All of these need to be deleted. You only need to change _existing_ `project()` calls. Please don't add any more.

**Review by ChrisThrasher — CHANGES_REQUESTED** [review]

File: moveit_core/collision_detection/CMakeLists.txt (hunk 1 above)
Comment (sameergupta4873) [reply moveit_core/collision_detection/CMakeLists.txt:4]: Got it, I will revert back with the changes

File: moveit_core/collision_detection/CMakeLists.txt [hunk 2]
```diff
@@ -31,7 +31,7 @@ ament_target_dependencies(moveit_collision_detection
 target_include_directories(moveit_collision_detection BEFORE PUBLIC $<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR}>)
 target_include_directories(moveit_collision_detection PUBLIC $<TARGET_PROPERTY:moveit_planning_scene,INCLUDE_DIRECTORIES>)
 
-set_target_properties(moveit_collision_detection PROPERTIES VERSION ${${PROJECT_NAME}_VERSION})
+set_target_properties(moveit_collision_detection PROPERTIES VERSION "${${PROJECT_NAME}_VERSION}")
```
Comment (ChrisThrasher) [inline moveit_core/collision_detection/CMakeLists.txt:34]:
```suggestion
set_target_properties(moveit_collision_detection PROPERTIES VERSION ${${PROJECT_NAME}_VERSION})
```

File: moveit_core/version/CMakeLists.txt [hunk 3]
```diff
@@ -1,5 +1,4 @@
 # Generate and install version.h
-
```
Comment (ChrisThrasher) [inline moveit_core/version/CMakeLists.txt:2]: Add this newline back

File: moveit_kinematics/cached_ik_kinematics_plugin/CMakeLists.txt [hunk 4]
```diff
@@ -19,7 +19,6 @@ generate_parameter_library(
   cached_ik_kinematics_parameters # cmake target name for the parameter library
   include/moveit/cached_ik_kinematics_plugin/cached_ik_kinematics_parameters.yaml # path to input yaml file
 )
-
```
Comment (ChrisThrasher) [inline moveit_kinematics/cached_ik_kinematics_plugin/CMakeLists.txt:22]: Add this newline back

File: moveit_runtime/CMakeLists.txt [hunk 5]
```diff
@@ -1,5 +1,5 @@
 cmake_minimum_required(VERSION 3.22)
-project(moveit_runtime)
+project(moveit_runtime LANGUAGES CXX)
```
Comment (ChrisThrasher) [inline moveit_runtime/CMakeLists.txt:2]:
```suggestion
project(moveit_runtime LANGUAGES NONE)
```

**Review by ChrisThrasher — CHANGES_REQUESTED** [review]

File: moveit_core/collision_detection/CMakeLists.txt (hunk 2 above)
Comment (ChrisThrasher) [reply moveit_core/collision_detection/CMakeLists.txt:34]: There are still more unresolved changes to make

**Review by ChrisThrasher — CHANGES_REQUESTED** [review]

**Comment by ChrisThrasher** [comment]
The `moveit_servo` package needs this change too.

**Review by ChrisThrasher — APPROVED** [review]

**Review by tylerjw — APPROVED** [review]
@ChrisThrasher thank you for doing the reviews here. @sameergupta4873 thank you for making this change! Welcome to contributing to MoveIt. :tada:

**Pull request merged**